server/src/metadata/field-metadata/field-metadata.entity.ts

```diff
@@ -11,29 +11,32 @@ import {
 } from 'typeorm';
 import {
   Authorize,
+  BeforeCreateOne,
   IDField,
   QueryOptions,
 } from '@ptc-org/nestjs-query-graphql';
 
 import { ObjectMetadata } from 'src/metadata/object-metadata/object-metadata.entity';
 
+import { BeforeCreateOneField } from './hooks/before-create-one-field.hook';
+
 export type FieldMetadataTargetColumnMap = {
   [key: string]: string;
 };
-
+@Entity('field_metadata')
 @ObjectType('field')
+@BeforeCreateOne(BeforeCreateOneField)
+@Authorize({
+  authorize: (context: any) => ({
+    workspaceId: { eq: context?.req?.user?.workspace?.id },
+  }),
+})
 @QueryOptions({
   defaultResultSize: 10,
```

Comment:
This could be extracted at the module level. However the package removed that ability a few patch earlier afaik, we can still extract this in a shared file anyway